Implement CI

reginleif888 opened the pull request.

reginleif888 committed:
Implement CI

reginleif888 committed:
fix CI

reginleif888 committed:
debug

reginleif888 committed:
fix test

reginleif888 committed:
Merge branch 'main' into implement-ci

reginleif888 committed:
fix

timohanke committed:
Use github actions to install mops, dfx

timohanke committed:
reorder ci commands

timohanke committed:
Revert "reorder ci commands"

This reverts commit 15b287b880dca33746d5e2d3231b97a242b22e35.

timohanke committed:
bump git action version numbers

timohanke commented:
Forgot to ask for checking did/directory.did file against the one generated by dfx build --check.

timohanke committed:
Reorder CI workflow

* Reorder to minimize dependencies of the tests clearer
* No need to run replica (dfx start) anymore

timohanke committed:
CI improvements (#11)

* Check did file with didc
* Remove dependency on internet_identity bindings in ci.yml
* Minimize ci.yml
* Check `internet_identity.did` into the repo and point dfx.json:
This seems to be the only way to make dfx pick up this entry in dfx.json:
```
"internet_identity": {
  "remote": {
     "candid": ".."
  }
}
```
Doing so eliminates the warning about did file not found.
